pkg/services/object/split/verify.go

```diff
@@ -0,0 +1,125 @@
+package split
+
+import (
+	"context"
+	"fmt"
+
+	getsvc "github.com/nspcc-dev/neofs-node/pkg/services/object/get"
+	cid "github.com/nspcc-dev/neofs-sdk-go/container/id"
+	"github.com/nspcc-dev/neofs-sdk-go/object"
+	oid "github.com/nspcc-dev/neofs-sdk-go/object/id"
+	"golang.org/x/sync/errgroup"
+)
+
+// NewVerifier returns Verifier that ready to Verifier.VerifySplit.
+// Get service must be non-nil, otherwise stable work is not
+// guaranteed.
+func NewVerifier(get *getsvc.Service) *Verifier {
+	return &Verifier{
+		get: get,
+	}
+}
+
+// Verifier implements [object.SplitVerifier] interface.
+type Verifier struct {
+	get *getsvc.Service
+}
+
+// VerifySplit verifies split chains:
+//  1. Children should be ordered correctly (in the sense of a linked list child objects compose)
+//  2. NeoFS should answer to the HEAD requests with children whose object size corresponds to the link's info
+//  3. Every child should have a provided first object in its split header (the first object itself is an exception)
+func (v *Verifier) VerifySplit(ctx context.Context, cnr cid.ID, firstID oid.ID, childrenFromLink []object.MeasuredObject) error {
+	// can be limited, depends on the expected big objects payload length
+	var wg errgroup.Group
+	receivedObjects := make([]*object.Object, len(childrenFromLink))
+
+	for i := range childrenFromLink {
+		iCopy := i
+		wg.Go(func() error {
+			var shouldHaveFirstObject *oid.ID
+			if iCopy != 0 {
+				shouldHaveFirstObject = &firstID
+			}
+
+			var err error
+			receivedObjects[iCopy], err = v.verifySinglePart(ctx, cnr, shouldHaveFirstObject, childrenFromLink[iCopy])
```

Comment:
1M * 16K for headers?

Reply:
how child objects may be 16K? optimizations may be presented here but not sure they changes smth for now (64mb are required for the most optimal part of the object storing if _NO_ get-check-drop is implemented and all the objects are fetched before any validation is being done)

Reply:
Fetch and verify group by group, headers are headers and can be up to 16KB.

Reply:
This is still a problem, but can be left for another issue.